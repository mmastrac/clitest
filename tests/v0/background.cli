#!/usr/bin/env clitest --v0

using tempdir;

set TEMP_FILE background.log;

background {
    $ printf "background\n" >> $TEMP_FILE && sleep 10 && printf "background\n" >> $TEMP_FILE
    %EXIT any
}

$ sleep .2 && cat $TEMP_FILE
repeat {
    ! background
}
