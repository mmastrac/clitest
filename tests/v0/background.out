Running tests/v0/background.cli ...

using tempdir: <tmp>

set TEMP_FILE background.log

sleep .2 && cat $TEMP_FILE
---
background
---
✅ OK

(background) printf "background\n" >> $TEMP_FILE && sleep 10 && printf "background\n" >> $TEMP_FILE
---
(no output)
---
✅ OK (signal: 1 (SIGHUP))

(cleanup) removing <tmp> && cd <root>

tests/v0/background.cli PASSED
1 test(s) passed
